packages/components/src/styles/components/dialog-primitive.scss

```diff
@@ -0,0 +1,146 @@
+}
+
+
+// Note: this wraps both .hds-dialog-primitive__header & .hds-dialog-primitive__description
+.hds-dialog-primitive__wrapper-header {
+  flex: none;
+  background: var(--token-color-surface-faint);
+  border-bottom: 1px solid var(--token-color-border-primary);
+  border-top-left-radius: inherit;
+  border-top-right-radius: inherit;
+
+  // because of the way in which Ember generates whitespace, we need to use this complicated selector to hide the container when it's "empty" (no children)
+  &:not(:has( > *)) {
+    display: none;
+  }
+}
+
+.hds-dialog-primitive__wrapper-body {
+  flex: 1 1 auto;
+  // this is necessary, otherwise if there are words longer than the width of the available space for container it will scroll horizontally (consumers will need to handle breaking words for this content, if they want)
+  overflow-x: hidden;
+  // we want the content to vertically scroll if needed
+  overflow-y: auto;
+  overscroll-behavior: contain;
+}
+
+.hds-dialog-primitive__wrapper-footer {
+  flex: none;
+  border-top: 1px solid var(--token-color-border-primary);
+  border-bottom-right-radius: inherit;
+  border-bottom-left-radius: inherit;
+
+  // because of the way in which Ember generates whitespace, we need to use this complicated selector to hide the container when it's "empty" (no children)
+  &:not(:has( > *)) {
+    display: none;
+  }
+}
+
+
+// HEADER
+
+.hds-dialog-primitive__header {
+  display: flex;
+  gap: 16px;
+  align-items: flex-start;
+  padding: $hds-dialog-primitive-vertical-padding $hds-dialog-primitive-horizontal-padding;
+  color: var(--token-color-foreground-strong); // default color (applied to both the title and the icon)
+  border-top-left-radius: inherit;
+  border-top-right-radius: inherit;
+}
+
+.hds-dialog-primitive__icon {
+  flex: none;
+  align-self: center;
+}
+
+.hds-dialog-primitive__title {
+  flex-grow: 1;
+}
+
+.hds-dialog-primitive__tagline {
+  margin-bottom: 4px;
+  color: var(--token-color-foreground-faint); // default color
+}
+
+.hds-dialog-primitive__dismiss {
+  align-self: flex-start;
+}
+
+
+// DESCRIPTION
+// (currently used only in Flyout, not in Modal)
+
+.hds-dialog-primitive__description {
+  padding: 0 $hds-dialog-primitive-horizontal-padding $hds-dialog-primitive-vertical-padding;
+}
+
+
+// BODY
+
+.hds-dialog-primitive__body {
+  padding: $hds-dialog-primitive-horizontal-padding;
+}
+
+
+// FOOTER
+
+.hds-dialog-primitive__footer {
+  padding: $hds-dialog-primitive-vertical-padding $hds-dialog-primitive-horizontal-padding;
+  background: var(--token-color-surface-faint);
+  border-bottom-right-radius: inherit;
+  border-bottom-left-radius: inherit;
+
+  // Tertiary buttons must always be placed/aligned at the end of the row
+  .hds-button-set {
+    .hds-button--color-tertiary {
+      margin-left: auto;
+    }
+  }
+}
+
+// OVERLAY
+
+.hds-dialog-primitive__overlay {
+  position: fixed;
+  top: 0;
+  right: 0;
+  bottom: 0;
+  left: 0;
```

Comment:
Could replace top, right, etc with `inset: 0` to set all 4 positions at once.

https://developer.mozilla.org/en-US/docs/Web/CSS/inset

Reply:
Yes, good idea. Done.